features/rolesandpermissions/impl/src/main/kotlin/io/element/android/features/rolesandpermissions/impl/roles/ChangeRolesPresenter.kt

```diff
@@ -214,9 +217,13 @@ class ChangeRolesPresenter(
                 saveState.value = AsyncAction.Failure(it)
             }
             .onSuccess {
-                saveState.value = AsyncAction.Success(true)
                 // Asynchronously reload the room members
-                launch { room.updateMembers() }
+                launch {
+                    withContext(NonCancellable) {
```

Comment:
Are you sure about this? It can be dangerous in rooms with lots of members, since it'll be a long running task, and we may destroy the `room` in the meantime.